packages/flutter/lib/src/foundation/math.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Same as [num.clamp] but optimized for non-null [double].
```

Comment:
nit: `non-null [double]s`.

It would be good to add some more information to this doc-comment, like why we're doing it and when it could be deprecated. I'd probably remove the 4x from the public facing doc-comment and instead just explain that this avoids boxing and polymorphism which make `num.clamp` slower as of right now.

I think I'd also add a non-doc comment explaining what benchmark verifies that this method is faster. We should add that benchmark and make it do something loud (fail?) if clampDouble is no longer faster than num.clamp. The non-doc comment can say something like `as of May 2022 this is 4x faster than num.clamp`.

At some point, num.clamp may get faster than this and we should know when it's worth switching back/deprecating this method.